client/source/js/modules/admin/manage-projects-ctrl.js

```diff
@@ -0,0 +1,125 @@
+define(['./module', 'angular', 'underscore'], function (module, angular, _) {
+  'use strict';
+
+  module.controller('AdminManageProjectsController', function ($scope, $http, info, projects, users, activeProject, UserManager, modalService) {
+    $scope.users = users.data.users;
+    $scope.users = _.compact(_.map(_(projects.data.projects).groupBy(function (p) {return p.user_id;}), function (projects, userId) {
+      var user = _.findWhere($scope.users, {id: parseInt(userId)});
+      return user.email===UserManager.data.email ? undefined :{
+        projects: projects,
+        data: user
+      };
+    }));
+
+    /**
+     * Regenerates workbook for the given project `name`
+     * Alerts the user if it cannot do it.
+     *
+     */
+    $scope.workbook = function (name, id) {
+      // read that this is the universal method which should work everywhere in
+      // http://stackoverflow.com/questions/24080018/download-file-from-a-webapi-method-using-angularjs
+      window.open('/api/project/workbook/' + id, '_blank', '');
+    };
+
+    /**
+     * Opens to edit an existing project using `name` in /project/create screen
+     *
+     * Alerts the user if it cannot do it.
+     */
+    $scope.edit = function (name, id) {
+      $http.get('/api/project/open/' + id)
+        .success(function (response) {
+          if (response && response.status === 'NOK') {
+            modalService.inform(undefined, undefined, response.reason);
+            return;
+          }
+          activeProject.setActiveProjectFor(name, id, UserManager.data);
+          window.location = '/#/project/edit';
+        });
+    };
+
+    /**
+     * Opens an existing project using `name`
+     *
+     * Alerts the user if it cannot do it.
+     */
+    $scope.open = function (name, id) {
+      $http.get('/api/project/open/' + id)
+        .success(function (response) {
+          if (response && response.status === 'NOK') {
+            modalService.inform(undefined, undefined, response.reason);
+            return;
+          }
+          activeProject.setActiveProjectFor(name, id, UserManager.data);
+          window.location = '/';
```

Comment:
Instead of window.location please use angular's $location. As far as I know it's doing the same thing, but the upside is that in case of testing you can mock $location.